pkg/retrieval/retrieval_test.go

```diff
@@ -0,0 +1,130 @@
+// Copyright 2020 The Swarm Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package retrieval_test
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io/ioutil"
+	"testing"
+	"time"
+
+	"github.com/ethersphere/bee/pkg/logging"
+	"github.com/ethersphere/bee/pkg/p2p/mock"
+	"github.com/ethersphere/bee/pkg/p2p/protobuf"
+	"github.com/ethersphere/bee/pkg/retrieval"
+	storemock "github.com/ethersphere/bee/pkg/storage/mock"
+)
+
+type mockPeerSuggester struct {
```

Comment:
I would move helper types to the bottom of the file in order to read the logic of the test first, then to see what is required for it to be executed. Nice mock, btw, it would be good that it would allow testing for errors also, something like pingpong/mock provides, but it is not crucial now.